hal/armv7m/imxrt/117x/imxrt.c

```diff
@@ -562,33 +592,51 @@ void _imxrt_init(void)
 	if (*(imxrt_common.stk + stk_ctrl) & 1)
 		*(imxrt_common.stk + stk_ctrl) &= ~1;
 
+	/* Disable deep sleep */
+	*(imxrt_common.scb + scb_scr) &= ~(1 << 2);
+	imxrt_dataBarrier();
+
+	/* Don't gate clocks on WFI */
+	for (i = pctl_lpcg_m7; i <= pctl_lpcg_uniq_edt_i; ++i)
+		_imxrt_setLevelLPCG(i, 4);
+
+	_imxrt_setDirectLPCG(pctl_lpcg_m7, 1);
+	_imxrt_setDirectLPCG(pctl_lpcg_sim_m7, 1);
+	//_imxrt_setDirectLPCG(pctl_lpcg_sim_m, 3);
+	//_imxrt_setDirectLPCG(pctl_lpcg_sim_disp, 3);
+	//_imxrt_setDirectLPCG(pctl_lpcg_sim_per, 3);
+	//_imxrt_setDirectLPCG(pctl_lpcg_sim_lpsr, 3);
+	//_imxrt_setDirectLPCG(pctl_lpcg_sim_r, 3);
+	//_imxrt_setDirectLPCG(pctl_lpcg_sim1, 3);
+	//_imxrt_setDirectLPCG(pctl_lpcg_sim2, 3);
+	_imxrt_setDirectLPCG(pctl_lpcg_jtag_mux, 1);
+	_imxrt_setDirectLPCG(pctl_lpcg_semc, 0);
+
 	/* Configure cache */
 	_imxrt_enableDCache();
 	_imxrt_enableICache();
 
+	/* gpr37 enable DBG_EN | NIDEN */
+	//*(imxrt_common.iomuxc_gpr + 37) |= 3;
+
 	/* Reconfigure all IO pads as slow slew-rate and low drive strength */
 	for (i = pctl_pad_gpio_emc_b1_00; i <= pctl_pad_gpio_disp_b2_15; ++i) {
 		if ((reg = _imxrt_IOpadGetReg(i)) == NULL)
 			continue;
 
 		*reg |= 1 << 1;
+		imxrt_dataBarrier();
 	}
 
 	for (; i <= pctl_pad_gpio_lpsr_15; ++i) {
 		if ((reg = _imxrt_IOpadGetReg(i)) == NULL)
 			continue;
 
 		*reg &= ~0x3;
+		imxrt_dataBarrier();
 	}
 
-	/* Allow userspace applications to access hardware registers */
-	/*
-	for (i = 0; i < sizeof(imxrt_common.aips) / sizeof(imxrt_common.aips[0]); ++i) {
-		*(imxrt_common.aips[i] + aipstz_opacr) &= ~0x44444444;
-		*(imxrt_common.aips[i] + aipstz_opacr1) &= ~0x44444444;
-		*(imxrt_common.aips[i] + aipstz_opacr2) &= ~0x44444444;
-		*(imxrt_common.aips[i] + aipstz_opacr3) &= ~0x44444444;
-		*(imxrt_common.aips[i] + aipstz_opacr4) &= ~0x44444444;
-	}
-*/
+	/* Enable UsageFault, BusFault and MemManage exceptions */
+	*(imxrt_common.scb + scb_shcsr) |= (1 << 16) | (1 << 17) | (1 << 18);
+	imxrt_dataBarrier();
```

Comment:
Additionally to enabled exceptions I would enable [DIV_0_TRP](https://developer.arm.com/documentation/dui0646/c/cortex-m7-peripherals/system-control-block/configuration-and-control-register) in SCB->CCR , what do You think ?

Reply:
Maybe in the future, I want to reduce number of changes to the minimum here